articles/manual/statements.md

```diff
@@ -0,0 +1,928 @@
+
+A mutable binding statement consists of the keyword `mutable`, followed by
+a symbol or symbol tuple, an equals sign `=`, an expression to bind the symbol(s) to, and a terminating semicolon.
+The type of the bound symbol(s) is inferred based on the expression on the right hand side. If a symbol is rebound later in the code, its type does not change, and the bound value needs to be compatible with that type.
+
+Mutability within Q# is a concept that applies to a *symbol* rather than a type or value. 
+Specifically, it does not have a representation in the type system, implicitly or explicitly, and whether or not a binding is mutable (as indicated by the `mutable` keyword) or immutable (as indicated by `let`) does not change the type of the bound variable(s). 
+This provides an important way to isolate mutability inside specialized functions and operations.
+In particular, even though an adjoint specialization for an operation which uses a mutable variable cannot be auto-generated, auto-generation works fine for an operation calling a function which uses mutability.
+For this reason, it is a good practice to make functions and operations which use mutability as short and compact as possible, so that the rest of the quantum program can be written using ordinary immutable variables.
+
+### Rebinding of Mutable Symbols
+
+A mutable variable may be rebound using a `set` statement.
+Such a rebinding consists of the keyword `set`, followed by
+a symbol or symbol tuple, an equals sign `=`, an expression to rebind the symbol(s) to, and a terminating semicolon.
+The value must be compatible with the type(s) of the symbol(s) it is bound to.
+
+#### Apply-and-Reassign Statement
+
+A particular kind of set-statement we refer to as an apply-and-reassign statement provides a convenient way of concatenation if the right hand side consists of the application of a binary operator and the result is to be rebound to the left argument to the operator. 
+For example,
+```qsharp
+mutable counter = 0;
+for (i in 1 .. 2 .. 10) {
+    set counter += 1;
+    // ...
+}
+```
+increments the value of the counter `counter` in each iteration of the `for` loop. The code above is equivalent to 
+```qsharp
+mutable counter = 0;
+for (i in 1 .. 2 .. 10) {
+    set counter = counter + 1;
+    // ...
+}
+```
+Similar statements are available for all binary operators in which the type of the left-hand-side matches the expression type. 
+This provides for example a convenient way to accumulate values:
+```qsharp
+mutable results = new Result[0];
+for (qubit in qubits) {
+    set results += [M(q)];
+    // ...
+}
+```
+
+#### Update-and-Reassign Statement
+
+A similar concatenation exists for copy-and-update expressions on the right hand side. 
+Correspondingly, update-and-reassign statements exist for named items in user-defined types as well as for array items.  
+
+```qsharp
+newtype Complex = (Re : Double, Im : Double);
```

Comment:
I think the named items in this don't quite correspond to Microsoft.Quantum.Math.Complex?